### PR TITLE
Add TimeInterval operator to Observable

### LINKED ISSUE #143: TimeInterval operator
`TimeInterval` operator implementation

http://reactivex.io/documentation/operators/timeinterval.html

### PR DESCRIPTION
**Pull request by diogobeda**
Hi @teivah, this is my first stab at contributing to a Go project and I'm just getting started with Go.
From what I've seen of the other implementations this one seems right, but I might have missed something.

Could you please review my code so I can then add docs to the operator? :)

Closes #143

### CODE REVIEW

**Comment by diogobeda** [comment]
The tests are a bit non deterministic. Will take a look at them when possible

File: observable.go [hunk 1]
```diff
@@ -1601,6 +1602,29 @@ func (o *observable) TakeWhile(apply Predicate) Observable {
...
 }
 
+func (o *observable) TimeInterval() Observable {
+	f := func(out chan interface{}) {
+		it := o.Iterator(context.Background())
+		var latestTime time.Time
+		for {
+			if _, err := it.Next(context.Background()); err == nil {
+				if latestTime.IsZero() {
+					out <- time.Duration(0)
```
Comment (teivah) [inline observable.go]: I'm not sure about sending this very item. According to the [documentation](http://reactivex.io/documentation/operators/timeinterval.html), the first item sent should rather be the duration between the subscription and the first item received.
In this regards, we should rather initialize `latestTime` to `time.Now()` in my opinion and get rid of this if/else statement.
What do you think?

Comment (teivah) [reply observable.go]: This should impact the tests as well.

**Comment by teivah** [comment]
> The tests are a bit non deterministic. Will take a look at them when possible

Do you know which one is failing from time to time? I already commented some of them using a `// FIXME comment`. I'm gonna create issues to track them.

File: observable.go (hunk 1 above)
Comment (diogobeda) [reply observable.go]: Nice, thanks for the great feedback @teivah. I wasn't sure how to represent the time of the subscription, but it's clearer now. Will fix as soon as I can.

**Comment by teivah** [comment]
Closed for no activity

**Pull request closed without merging**